api/api.go

```diff
@@ -316,6 +331,10 @@ func (c *Client) GetNodeClient(nodeID string, q *QueryOptions) (*Client, error)
 		region = q.Region
 	}
 
+	if region == "" {
```

Comment:
This logic is a bit confusing with 329-332. Is there a way (case statement or "else if") to have region only be assigned a value once?